File: merge_all_remaining.py
#!/usr/bin/env python3

import os
import subprocess
import json
import time

def run_command(cmd, timeout=60):
    """Run a command with timeout and return result"""
    try:
        result = subprocess.run(cmd, shell=True, capture_output=True, text=True, timeout=timeout)
        return result.returncode, result.stdout, result.stderr
    except subprocess.TimeoutExpired:
        print(f"Command timed out: {cmd}")
        return -1, "", "Timeout"
    except Exception as e:
        print(f"Command failed: {cmd}, Error: {e}")
        return -1, "", str(e)

def main():
    print("🚀 Processing all remaining PRs in batches...")
    
    # Change to workspace directory
    os.chdir('/workspace')
    
    # Step 1: Ensure we're on main branch and up to date
    print("Step 1: Ensuring main branch is up to date...")
    run_command("git checkout main")
    run_command("git pull origin main")
    
    total_successful = 0
    total_failed = 0
    batch_num = 1
    
    while True:
        print(f"\n--- Processing Batch {batch_num} ---")
        
        # Get next batch of branches (starting from where we left off)
        offset = (batch_num - 1) * 50 + 50  # Start from 50, 100, 150, etc.
        code, stdout, stderr = run_command(f"git branch -r | grep 'fix-netlify-build-and-merge-to-main' | head -{offset} | tail -50")
        
        if code != 0 or not stdout.strip():
            print("No more branches found")
            break
        
        branches = stdout.strip().split('\n')
        print(f"Found {len(branches)} branches in batch {batch_num}")
        
        if len(branches) == 0:
            break
        
        # Try to merge each branch
        successful_merges = 0
        failed_merges = 0
        
        for i, branch in enumerate(branches):
            branch = branch.strip().replace('origin/', '')
            if not branch:
                continue
                
            print(f"[Batch {batch_num}, {i+1}/{len(branches)}] Processing: {branch}")
            
            # Try to merge the branch
            code, stdout, stderr = run_command(f"git merge origin/{branch} --no-edit")
            
            if code == 0:
                print(f"✅ Successfully merged {branch}")
                successful_merges += 1
                total_successful += 1
            else:
                print(f"❌ Failed to merge {branch}: {stderr}")
                failed_merges += 1
                total_failed += 1
                
                # Try to abort the merge if it failed
                run_command("git merge --abort")
        
        # Push changes if any merges were successful
        if successful_merges > 0:
            print(f"\nPushing {successful_merges} successful merges from batch {batch_num}...")
            code, stdout, stderr = run_command("git push origin main")
            if code == 0:
                print("✅ Successfully pushed to origin/main")
            else:
                print(f"❌ Failed to push: {stderr}")
        
        print(f"Batch {batch_num} Summary: {successful_merges} successful, {failed_merges} failed")
        
        # If no successful merges in this batch, we might be done
        if successful_merges == 0:
            print("No successful merges in this batch, checking if we're done...")
            break
        
        batch_num += 1
        
        # Safety limit to prevent infinite loops
        if batch_num > 50:
            print("Reached safety limit of 50 batches")
            break
    
    print(f"\n🎉 FINAL SUMMARY:")
    print(f"   Total successful merges: {total_successful}")
    print(f"   Total failed merges: {total_failed}")
    print(f"   Total processed: {total_successful + total_failed}")
    print(f"   Batches processed: {batch_num - 1}")

if __name__ == "__main__":
    main()

File: merge-all-remaining.js
#!/usr/bin/env node
/**
 * Merge All Remaining Branches - Comprehensive merge of all remaining branches
 */ import { execSync } from 'child_process'
import fs from 'fs'
// //Step 1: Ensure we're on main and up to date,
// try {,
  execSync('git checkout main', { stdio: 'inherit' })
  execSync('git pull origin main', { stdio: 'inherit' })
//   } catch (error) {
    //   process.exit(1)
  }

//Step 2: Get all remaining branches
// //Get all remote branches,
const allBranches = execSync('git branch -r', { encoding: 'utf8' })
// //Step,
  1: Ensure we're on main and up to date,
// try {/* TODO: Fix JSX expression */}
  o: 'inherit' })
  execSync('git pull origin main', {/* TODO: Fix JSX expression */})
  o: 'inherit' })
//   } catch (error) {/* TODO: Fix JSX expression */}
}

//Step,
  2: Get all remaining branches
// //Get all remote branches,
const allBranches = execSync('git branch -r', {/* TODO: Fix JSX expression */})
  g: 'utf8' })
  .split('\n')
  .map(branch => branch.trim())
  .filter(branch => branch && !branch.includes('HEAD'))
  .filter(branch => !branch.includes('backup'))
  .filter(branch => !branch.includes('aggressive'))
  .filter(branch => !branch.includes('automation'))
  .map(branch => branch.replace('origin/', ''))
//Filter for relevant branches
const relevantBranches = allBranches.filter(branch =>)
    branch.includes('cursor') ||
    branch.includes('fix') ||
    branch.includes('merge') ||
    branch.includes('candidate') ||
    branch.includes('chore') ||
    branch.includes('add-new') ||
    branch.includes('ai-')
)
// //Step 3: Merge function with conflict resolution,
function mergeBranch(branchName) {,
//   try {,
    //Check if branch exists
    execSync(`git fetch origin ${branchName}`, { stdio: 'pipe' })
    //Check if already merged
    const isMerged = execSync(`git branch --merged main | grep -q "${branchName}" || echo "not_merged"`)
      { encoding: 'utf8' }
// //Step,
  3: Merge function with conflict resolution,
function mergeBranch(branchName) {/* TODO: Fix JSX expression */}
    execSync(`git fetch origin ${branchName}`, {/* TODO: Fix JSX expression */})
  o: 'pipe' })
    //Check if already merged;`
    const isMerged = execSync(`git branch --merged main | grep -q "${branchName}" || echo "not_merged"`,
      {/* TODO: Fix JSX expression */}
  g: 'utf8' })
    ).trim()
    if (isMerged !== 'not_merged') {/* TODO: Fix JSX expression */}
  d: 'already_merged' }
    }

    //Try to merge
    try {
      execSync(`git merge origin/${branchName} --no-ff -m "Merge ${branchName}: automated merge"`)
        { stdio: 'inherit' }
      )
//       return { success: true, method: 'direct' }
    } catch (mergeError) {
//       //Try different conflict resolution strategies
      try {
        //Strategy 1: Use theirs,
        execSync(`git merge origin/${branchName} --strategy-option=theirs --no-ff -m "Merge ${branchName}: using theirs strategy"`)
          { stdio: 'inherit' }
        )
//         return { success: true, method: 'theirs' }
      } catch (theirsError) {
        try {
          //Strategy 2: Use ours,
          execSync(`git merge origin/${branchName} --strategy-option=ours --no-ff -m "Merge ${branchName}: using ours strategy"`)
            { stdio: 'inherit' }
    try {/* TODO: Fix JSX expression */}"`
        `git merge origin/${branchName} --no-ff -m "Merge ${branchName}: automated merge"`,
        {/* TODO: Fix JSX expression */}
  o: 'inherit' }
      )
//       return {/* TODO: Fix JSX expression */}
  d: 'direct' }
    } catch (mergeError) {/* TODO: Fix JSX expression */}"`
          `git merge origin/${branchName} --strategy-option=theirs --no-ff -m "Merge ${branchName}: using theirs strategy"`,
          {/* TODO: Fix JSX expression */}
  o: 'inherit' }
        )
//         return {/* TODO: Fix JSX expression */}
  d: 'theirs' }
      } catch (theirsError) {/* TODO: Fix JSX expression */}"`
            `git merge origin/${branchName} --strategy-option=ours --no-ff -m "Merge ${branchName}: using ours strategy"`,
            {/* TODO: Fix JSX expression */}
  o: 'inherit' }
          )
//           return {/* TODO: Fix JSX expression */}
  d: 'ours' }
        } catch (oursError) {/* TODO: Fix JSX expression */}
  d: 'failed' }
        }
      }
    }
  } catch (error) {/* TODO: Fix JSX expression */}
  d: 'not_found' }
  }
}

//Step 4: Process branches in batches
const results = {
  successful: []
  failed: []
  summary: {
    total: 0
    successful: 0
    failed: 0
    methods: {
      direct: 0
      theirs: 0
      ours: 0
      already_merged: 0
      not_found: 0
      failed: 0,
//Step,
  4: Process branches in batches,
const results = {/* TODO: Fix JSX expression */};
    }}}

// //Process in batches of 50 to avoid overwhelming the system
// const batchSize = 50
// const totalBatches = Math.ceil(relevantBranches.length / batchSize)
for (let batch = 0; batch < totalBatches; batch++) {/* TODO: Fix JSX expression */}`
    `\n📦 Processing batch ${batch + 1}/${totalBatches} (${batchBranches.length} branches)...`
  )
  for (const branch of batchBranches) {
    results.summary.total++
    if (result.success) {
      results.successful.push({)
        branch: branch,),
        success: true),
        method: result.method)})
      results.summary.successful++
      results.summary.methods[result.method]++
    } else {
      results.failed.push({)
        branch: branch,),
        success: false),
        method: result.method),
  for (const branch of batchBranches) {/* TODO: Fix JSX expression */}
      })
      results.summary.successful++
      results.summary.methods[result.method]++
    } else {/* TODO: Fix JSX expression */}
      })
      results.summary.failed++
      results.summary.methods[result.method]++
    }
  }

  //Push changes after each batch
  if (batch % 5 === 0 || batch === totalBatches - 1) {
    try {
      execSync('git push origin main', { stdio: 'inherit' })
//       } catch (error) {
  if (batch % 5 === 0 || batch === totalBatches - 1) {/* TODO: Fix JSX expression */}
  o: 'inherit' })
//       } catch (error) {/* TODO: Fix JSX expression */}
//       }
  }
}

//Step 5: Generate final report,
// const report = {,
  ...results
  timestamp: new Date().toISOString()}
;
fs.writeFileSync('all-remaining-branches-merge-report.json')
  JSON.stringify(report, null, 2)
)
//Step 6: Final push,
// try {,
  execSync('git push origin main', { stdio: 'inherit' })
//   } catch (error) {
//   }

// Step 7: Summary,
// // // // // // // // // // // // // // ,
//Step,
  5: Generate final report,
// const report = {/* TODO: Fix JSX expression */};
}

fs.writeFileSync('all-remaining-branches-merge-report.json')
  JSON.stringify(report, null, 2)
)
//Step,
  6: Final push,
// try {/* TODO: Fix JSX expression */}
  o: 'inherit' })
//   } catch (error) {/* TODO: Fix JSX expression */}
//   }

// Step,
  7: Summary,
// // // // // // // // // // // // // // "`
